yocto-update-strategies.md

```diff
@@ -0,0 +1,651 @@
+    able to compile it manually
+
+* defconfig
+
+.code-14px[```bash
++ CONFIG_EXPERT=y
++ CONFIG_GPIO_SYSFS=y
+```]
+
+???
+
+CONFIG_GPIO_SYSFS=y - enabling EXPERT in order to get the same config as before
+an update
+
+---
+# Build troubleshooting - kernel
+
+* device-tree I
+
+.code-14px[```bash
++	assigned-clocks = <&clk IMX8MM_CLK_PCIE1_AUX>,
++			  <&clk IMX8MM_CLK_PCIE1_PHY>,
++			  <&clk IMX8MM_CLK_PCIE1_CTRL>;
++	assigned-clock-rates = <10000000>, <100000000>, <250000000>;
++	assigned-clock-parents = <&clk IMX8MM_SYS_PLL2_50M>,
++				 <&clk IMX8MM_SYS_PLL2_100M>,
++				 <&clk IMX8MM_SYS_PLL2_250M>;
+```]
+
+* device-tree II
+
+.code-14px[```bash
+&sai2 {
+        pinctrl-names = "default";
+        pinctrl-0 = <&pinctrl_sai2>;
+        pinctrl-1 = <&pinctrl_codec_en>;
+-       clocks = <&clk IMX8MM_CLK_SAI2_ROOT>;
+        assigned-clocks = <&clk IMX8MM_CLK_SAI2>;
+        assigned-clock-parents = <&clk IMX8MM_AUDIO_PLL1_OUT>;
+        assigned-clock-rates = <24576000>;
+        (...)
+```]
+
+???
+
+Add `assigned-clocks`, `assigned-clock-rates` and
+`assigned-clock-parents` nodes to `pcie0`.
+
+mimics
+https://gitlab.com/3mdeb/dave_eu/linux-imx/-/commit/0d5920e4c35d15a9999ee19bb0521f21158072a0
+Specify the clock rate and parent of i.MX8MQ/MM PCIe clocks.
+
+sai2 overwriting correctly setting clocks in other file which caused whole
+platform hang while booting
+
+---
+# Build troubleshooting - default applications
+
+* Default applications as a part of created image
+  - by default we mean here part of systems to which we did not provide custom,
+    Client's source code
+  - issues here: configuration changes, compilation errors
+
+* Configuration changes
+  - missed or changed PACKAGECONFIG
+  - custom `do_configure` functions
+
+.code-14px[```bash
+- PACKAGECONFIG += "ssl libssh2"
++ PACKAGECONFIG += "openssl libssh2"
+```]
+
+tensorflow https://github.com/nxp-imx/meta-imx/pull/16
+
+---
+# Build troubleshooting - default applications
+
+* Compilation errors
+
+.code-14px[```bash
+/tmp/(...)/python/interpreter_wrapper/interpreter_wrapper_pybind11.cc
+| /work/build-wayland/tmp/(...)/python/interpreter_wrapper/interpreter_wrapper_pybind11.cc:16:10:
+    fatal error: pybind11/functional.h: No such file or
+directory
+| 16 | #include "pybind11/functional.h"
+(...)
+| In file included from /work/build-wayland/tmp/(...)/interpreter_wrapper.cc:15:
+| /work/build-wayland/tmp/(...)/python/interpreter_wrapper/interpreter_wrapper.h:28:10:
+    fatal error: Python.h: No such file or directory
+| 28 | #include <Python.h>
+| | ^~~~~~~~~~
+| compilation terminated.
+| ninja: build stopped: subcommand failed.
+```]
+
+---
+# Build troubleshooting - custom applications
+
+* Custom applications often are provided by our Clients
+  - they were compiled in some environment
+  - we integrate them for given Yocto release
+  - update may introduce problems (compilator flags, deprecated variables)
```

Comment:
shorten so it fits in one line. This way the code will better fit in the slide as well

Reply:
done https://github.com/3mdeb/conferences/pull/7/commits/471facce24fae0adab35f05332fba3bfa4009929